pkg/memory-monitor/src/statistics-collector/main.c

```diff
@@ -0,0 +1,108 @@
+#include <fcntl.h>
+#include <limits.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <time.h>
+#include <unistd.h>
+
+#define BUFFER_SIZE 256
+#define OUTPUT_FILE "psi_stats.txt"
+#define DEVICE_OUTPUT_DIR "/persist/memory-monitor/output"
+#define PROC_FILE "/proc/pressure/memory"
+
+void write_to_file(int fd, const char *buffer) {
+    if (write(fd, buffer, strlen(buffer)) == -1) {
+        perror("write");
+        exit(EXIT_FAILURE);
+    }
+    if (fsync(fd) == -1) {
+        perror("fsync");
+        exit(EXIT_FAILURE);
+    }
+}
+
+static void daemonize(void)
+{
+    int pid, sid;
+    pid = fork();
+    if (pid < 0) {
+        exit(EXIT_FAILURE);
+    }
+    // If we got a good PID, then we can exit the parent process
+    if (pid > 0) {
+        exit(EXIT_SUCCESS);
+    }
+
+    // Change the file mode mask
+    umask(0);
+
+    // Create a new SID for the child process
+    sid = setsid();
+    if (sid < 0) {
+        exit(EXIT_FAILURE);
+    }
+}
+
+int main()
+{
+    int fd;
+    FILE *proc_file;
+    char buffer[BUFFER_SIZE];
+    time_t rawtime;
+    struct tm *timeinfo;
+
+    daemonize();
+
+
+    // If /persist directory exists, create the output file there, otherwise create it in the current directory
+    // But first, create the filename dynamically, in a static size buffer
+    char output_file[PATH_MAX];
+    if (access(DEVICE_OUTPUT_DIR, F_OK) == 0) {
+        snprintf(output_file, PATH_MAX, "%s/%s", DEVICE_OUTPUT_DIR, OUTPUT_FILE);
+    } else {
+        snprintf(output_file, PATH_MAX, "./%s", OUTPUT_FILE);
+    }
+
+    fd = open(output_file, O_WRONLY | O_CREAT | O_TRUNC, 0644);
```

Comment:
```suggestion
    fd = open(output_file, O_WRONLY | O_CREAT | O_TRUNC | O_DIRECT | O_SYNC, 0644);
```
so that you don't have to `fsync` all the time?

Reply:
True. Can use this approach.